libs/SalesforceSDK/src/com/salesforce/androidsdk/accounts/UserAccount.java

```diff
@@ -192,15 +206,21 @@ public class UserAccount {
 		this.nativeLogin = nativeLogin;
 		this.language = language;
 		this.locale = locale;
+		this.cookieClientSrc = cookieClientSrc;
+		this.cookieSidClient = cookieSidClient;
+		this.sidCookieName = sidCookieName;
+		this.clientId = clientId;
 		SalesforceSDKManager.getInstance().registerUsedAppFeature(Features.FEATURE_USER_AUTH);
 	}
 
 	/**
-	 * Parameterized constructor.
+	 * Construct UserAccount from JSON.
 	 *
 	 * @param object JSON object.
+	 * @param applicationName the application name.
+	 * @param additionalOauthKeys addtional auth keys.
 	 */
-	public UserAccount(JSONObject object) {
+	UserAccount(JSONObject object, String applicationName, List<String> additionalOauthKeys) {
```

Comment:
This constructor does not call methods SalesforceSDKManager (to get application name of additional oauth keys) - which makes testing easier.